### PR TITLE
Feature/abstraction and hooks

### PR DESCRIPTION
**Pull request by ian-emsens-shd**
- Ran `format:fix`
- Ran `extract-translations`
- Disabled one lint rule
- Adjust some node scripts
- Split a lot of reusable features into dedicated hooks
- Aligned Edit and Create assignment views

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -61,5 +61,7 @@ module.exports = {
 				ignoreDeclarationSort: true,
 			},
 		],
+
+		'react/prop-types': 'off'
```
Comment (bertyhell) [inline .eslintrc.js:65]: is dit iets dat we in de toekomst terug willen aanzetten? of heeft die typecheck geen nut?

File: src/assignment/hooks/assignment-blocks-list.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        listSorterItem?: Partial<ListSorterItem>
+    }
+) {
+	const [t] = useTranslation();
+
+	const items = useMemo(() => {
+		return assignment.blocks.map((block) => {
+			const mapped: AssignmentBlock & ListSorterItem = {
+				...block,
+                ...config?.listSorterItem,
```
Comment (bertyhell) [inline src/assignment/hooks/assignment-blocks-list.tsx:32]: use tabs instead of spaces

File: src/assignment/hooks/assignment-blocks-list.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				icon: ASSIGNMENT_CREATE_UPDATE_BLOCK_ICONS()[block.type],
+				onPositionChange: (item, delta) => {
+					const switched = switchAssignmentBlockPositions(assignment.blocks, item, delta);
+
+					console.info('event', switched);
```
Comment (bertyhell) [inline src/assignment/hooks/assignment-blocks-list.tsx:37]: remove console log

File: src/assignment/hooks/assignment-block-meta.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { AssignmentBlock } from '@viaa/avo2-types/types/assignment';
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import { APP_PATH } from '../../constants';
+import { SearchFilter } from '../../search/search.const';
+import { buildLink, formatDate } from '../../shared/helpers';
+
+export function useAssignmentBlockMeta() {
```
Comment (bertyhell) [inline src/assignment/hooks/assignment-block-meta.tsx:9]: wouldn't this be better as a component?
```
 <AssignmentBlockMeta>
```

This is how i usually decide between a custom hook and a component:
* if a custom hook only returns one ReactNode it is better suited to be a function component
* if a hook returns multiple variables or the return type if not a ReactNode's then a custom hook makes sense

File: src/assignment/views/AssignmentCreate.tsx [hunk 5]
```diff
@@ -265,9 +233,30 @@ const AssignmentCreate: FunctionComponent<DefaultSecureRouteProps> = ({ user, hi
...
+		});
+
+		trigger();
+	}, [assignment, setValue, trigger]);
+
+	// Set the loading state when the form is ready
+	useEffect(() => {
+		if (loadingInfo.state !== 'loaded') {
+			assignment && setLoadingInfo({ state: 'loaded' });
+		}
```
Comment (bertyhell) [inline src/assignment/views/AssignmentCreate.tsx:257]:
```suggestion
		if (loadingInfo.state !== 'loaded' && assignment) {
			setLoadingInfo({ state: 'loaded' });
		}
```

**Review by bertyhell — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (ian-emsens-shd) [reply .eslintrc.js:65]: Binnen TS geen nut https://reactjs.org/docs/typechecking-with-proptypes.html

**Comment by ian-emsens-shd** [comment]
Merging, changes to assignment-block-meta will be picked up in a future PR

**Pull request merged**